UnitTests/Onboarding/ContextualOnboarding/BrowserTabViewControllerOnboardingTests.swift

```diff
@@ -146,15 +257,42 @@ class CapturingDialogFactory: ContextualDaxDialogsFactory {
     var capturedType: ContextualDialogType?
     var capturedDelegate: OnboardingNavigationDelegate?
 
+    private var onGotItPressed: (() -> Void)?
+    private var onFireButtonPressed: (() -> Void)?
+
     init(expectation: XCTestExpectation) {
         self.expectation = expectation
     }
 
-    func makeView(for type: ContextualDialogType, delegate: OnboardingNavigationDelegate, onDismiss: @escaping () -> Void, onGotItPressed: @escaping () -> Void) -> AnyView {
+    func makeView(for type: ContextualDialogType, delegate: OnboardingNavigationDelegate, onDismiss: @escaping () -> Void, onGotItPressed: @escaping () -> Void, onFireButtonPressed: @escaping () -> Void) -> AnyView {
         capturedType = type
         capturedDelegate = delegate
+        self.onGotItPressed = onGotItPressed
+        self.onFireButtonPressed = onFireButtonPressed
         expectation.fulfill()
         return AnyView(OnboardingFinalDialog(highFiveAction: {}))
     }
 
+    func performOnGotItPressed() {
+        onGotItPressed?()
+    }
+
+    func performOnFireButtonPressed() {
+        onFireButtonPressed?()
+    }
+
+}
+
+final class BrowserTabViewControllerDelegateMock: BrowserTabViewControllerDelegate {
```

Comment:
nit I guess should be called BrowserTabViewControllerDelegateCapturing or CapturingBrowserTabViewControllerDelegate

since it only captures data and does not mock behaviour

Reply:
changed to `BrowserTabViewControllerDelegateSpy`